cw_dogecoin/lib/src/dogecoin_wallet.dart

```diff
@@ -0,0 +1,185 @@
+
+  static Future<DogeCoinWallet> create(
+      {required String mnemonic,
+      required String password,
+      required WalletInfo walletInfo,
+      required Box<UnspentCoinsInfo> unspentCoinsInfo,
+      required EncryptionFileUtils encryptionFileUtils,
+      String? passphrase,
+      String? addressPageType,
+      List<BitcoinAddressRecord>? initialAddresses,
+      ElectrumBalance? initialBalance,
+      Map<String, int>? initialRegularAddressIndex,
+      Map<String, int>? initialChangeAddressIndex}) async {
+    return DogeCoinWallet(
+      mnemonic: mnemonic,
+      password: password,
+      walletInfo: walletInfo,
+      unspentCoinsInfo: unspentCoinsInfo,
+      initialAddresses: initialAddresses,
+      initialBalance: initialBalance,
+      seedBytes: MnemonicBip39.toSeed(mnemonic, passphrase: passphrase),
+      encryptionFileUtils: encryptionFileUtils,
+      initialRegularAddressIndex: initialRegularAddressIndex,
+      initialChangeAddressIndex: initialChangeAddressIndex,
+      addressPageType: P2pkhAddressType.p2pkh,
+      passphrase: passphrase,
+    );
+  }
+
+  static Future<DogeCoinWallet> open({
+    required String name,
+    required WalletInfo walletInfo,
+    required Box<UnspentCoinsInfo> unspentCoinsInfo,
+    required String password,
+    required EncryptionFileUtils encryptionFileUtils,
+  }) async {
+    final hasKeysFile = await WalletKeysFile.hasKeysFile(name, walletInfo.type);
+
+    ElectrumWalletSnapshot? snp = null;
+
+    try {
+      snp = await ElectrumWalletSnapshot.load(
+        encryptionFileUtils,
+        name,
+        walletInfo.type,
+        password,
+        DogecoinNetwork.mainnet,
+      );
+    } catch (e) {
+      if (!hasKeysFile) rethrow;
+    }
+
+    final WalletKeysData keysData;
+    // Migrate wallet from the old scheme to then new .keys file scheme
+    if (!hasKeysFile) {
+      keysData =
+          WalletKeysData(mnemonic: snp!.mnemonic, xPub: snp.xpub, passphrase: snp.passphrase);
+    } else {
+      keysData = await WalletKeysFile.readKeysFile(
+        name,
+        walletInfo.type,
+        password,
+        encryptionFileUtils,
+      );
+    }
+
+    return DogeCoinWallet(
+      mnemonic: keysData.mnemonic!,
+      password: password,
+      walletInfo: walletInfo,
+      unspentCoinsInfo: unspentCoinsInfo,
+      initialAddresses: snp?.addresses.map((addr) {
+        try {
+          return BitcoinAddressRecord(
+            addr.address,
+            index: addr.index,
+            isHidden: addr.isHidden,
+            name: addr.name,
+            type: P2pkhAddressType.p2pkh,
+            network: DogecoinNetwork.mainnet,
+          );
+        } catch (_) {
+          return BitcoinAddressRecord(
+            addr.address,
+            index: addr.index,
+            isHidden: addr.isHidden,
+            name: addr.name,
+            type: P2pkhAddressType.p2pkh,
+            network: DogecoinNetwork.mainnet,
+          );
+        }
+      }).toList(),
```

Comment:
this was for bitcoin cash for a reason, but here I believe it can be just `snp?.addresses` like litecoin, right?